satellite.go

```diff
@@ -0,0 +1,218 @@
+		if err != nil {
+			return err.Error()
+		}
+		msg = string(data)
+	case "":
+		break
+	default:
+		msg = "error: unsupported document format: " + result.format
+	}
+	return msg
+}
+
+// SatelliteHTTPClient represents http httpClient used for communication with
+// satellite server
+type SatelliteHTTPClient struct {
+	httpClient   *http.Client
+	satelliteURL *url.URL
+}
+
+// NewSatelliteClient creates instance of SatelliteHTTPClient and
+// configure it to use HTTPS
+func NewSatelliteClient(satelliteURL *url.URL) *SatelliteHTTPClient {
+	satClient := SatelliteHTTPClient{}
+	// We have to use insecure HTTPs connection, because most of the customers use
+	// self-signed certificates
+	tlsConfig := tls.Config{
+		InsecureSkipVerify: true,
+	}
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = &tlsConfig
+	satClient.httpClient = &http.Client{Transport: transport}
+	satClient.satelliteURL = satelliteURL
+	return &satClient
+}
+
+// Ping tries to ping Satellite server to be sure that user
+// tries to reach get and run bootstrap script from Satellite server.
+// We use following endpoint that is available for unauthenticated users:
+// https://satellite.sat.engineering.redhat.com/apidoc/v2/ping/ping.en.html
+// It is not necessary to try to reach this endpoint, but download and run
+// some bash script downloaded from the internet. It is risky action. For
+// this reason we do this extra step to minimize risk that some non-intentional
+// code is downloaded and run as root user.
+//
+// In the future we could use following endpoint to determine version of
+// Satellite (Katello) and change rhc behavior according detected Satellite version
+// https://satellite.sat.engineering.redhat.com/apidoc/v2/ping/server_status.en.html
+func (client *SatelliteHTTPClient) Ping() (*SatellitePingResponse, error) {
+	// Copy URL from struct URL and modify scheme, path and query
+	satellitePingUrl := *client.satelliteURL
+	satellitePingUrl.Scheme = "https"
+	satellitePingUrl.Path = "/api/ping"
+	satellitePingUrl.RawQuery = ""
+
+	response, err := client.httpClient.Get(satellitePingUrl.String())
+	if err != nil {
+		return nil, fmt.Errorf("ping satellite server failed with error: %v", err)
+	}
+	if response.StatusCode != 200 {
+		return nil, fmt.Errorf("ping satellite server failed with status code %d", response.StatusCode)
+	}
+	defer func() {
+		// TODO: If error happens, then log this error
+		_ = response.Body.Close()
+	}()
+	resBody, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read body of ping satellite server response: %v", err)
+	}
+
+	satellitePingResponse := SatellitePingResponse{}
+	err = json.Unmarshal(resBody, &satellitePingResponse)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse ping satellite server response: %v", err)
+	}
+
+	return &satellitePingResponse, nil
+}
+
+// downloadSatelliteScript tries to download script from provided URL to given file
```

Comment:
This method documentation looks like a leftover from the previous implementation. It doesn't match the current method name and it describes a parameter that it no longer accepts.

```suggestion
// downloadScript tries to download script from provided URL to given file
```